string dino_begin
---
string full_dino